fluss-connectors/fluss-connector-flink/src/main/java/com/alibaba/fluss/connector/flink/sink/FlinkTableSink.java

```diff
@@ -281,6 +290,15 @@ private void validateUpdatableAndDeletable() {
                             "Table %s is a Log Table. Log Table doesn't support DELETE and UPDATE statements.",
                             tablePath));
         }
+
+        if (mergeEngine != null && mergeEngine == ConfigOptions.MergeEngine.FIRST_ROW) {
+            throw new UnsupportedOperationException(
+                    String.format(
+                            "Table %s is with merge engine '%s'. Table with '%s' merge engine doesn't support DELETE and UPDATE statements.",
```

Comment:
Table xxx uses the 'first_row' merge engine, which does not support DELETE or UPDATE statements.